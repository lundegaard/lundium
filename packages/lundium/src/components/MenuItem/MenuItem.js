import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

const MenuItem = ({ className, noGutters, ...rest }) => (
	<li
		className={cx(
			{
				menu__item: true,
				['menu__item--no-gutters']: noGutters,
			},
			className,
		)}
		{...rest}
	/>
);

MenuItem.propTypes = {
	className: PropTypes.string,
	noGutters: PropTypes.bool,
};

export default MenuItem;
