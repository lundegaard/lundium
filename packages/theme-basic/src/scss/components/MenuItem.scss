@import '../core';

/*
 -  Menu Item
 */


.menu__item {
	width: 100%;
	padding: logical ( $spacer * .5 ) $spacer;

	&--no-gutters {
		padding: 0;
	}

	&:hover:not([class*='disabled']),
	&:focus:not([class*='disabled']) {
		background-color: $theme-gray-100;
		cursor: pointer;
	}

	.form-control[type='checkbox'] + .control-label {
		text-transform: none;
		letter-spacing: normal;
		width: 100%;
	}
}