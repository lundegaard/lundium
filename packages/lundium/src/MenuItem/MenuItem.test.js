import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from './MenuItem';

describe('<MenuItem />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<MenuItem {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
