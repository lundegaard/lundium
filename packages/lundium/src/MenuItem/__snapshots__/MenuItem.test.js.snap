// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> matches snapshot when shallowly rendered 1`] = `
<li
  className="menu__item"
/>
`;
